alicloud/resource_alicloud_datahub_topic.go

```diff
@@ -0,0 +1,216 @@
+package alicloud
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aliyun/aliyun-datahub-sdk-go/datahub"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAlicloudDatahubTopic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunDatahubTopicCreate,
+		Read:   resourceAliyunDatahubTopicRead,
+		Update: resourceAliyunDatahubTopicUpdate,
+		Delete: resourceAliyunDatahubTopicDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDatahubProjectName,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.ToLower(new) == strings.ToLower(old)
+				},
+			},
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDatahubTopicName,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.ToLower(new) == strings.ToLower(old)
+				},
+			},
+			"shard_count": &schema.Schema{
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIntegerInRange(1, 256),
+			},
+			"life_cycle": &schema.Schema{
```

Comment:
I think lifecycle is a word.

In addition, if lifecycle's values are the specified meaningful string, it will be more understandable than number.